helper/helper_test.go

```diff
@@ -98,4 +265,235 @@ var _ = Describe("helper tests", func() {
 			Expect(isSubset).To(Equal(false))
 		})
 	})
+
+	Describe("doRestoreAgent Mocked unit tests", func() {
+		BeforeEach(func() {
+			// Setup mocked tests environment
+			*singleDataFile = false
+			*content = 1
+			*oidFile = "testoid.dat"
+			*isResizeRestore = true
+			*origSize = 5
+			*destSize = 3
+			*pipeFile = "mock"
+			*onErrorContinue = true
+		})
+		It("successfully restores using a single data file when inputs are valid and no errors occur", func() {
+			*singleDataFile = true
+
+			oidBatch := []oidWithBatch{{oid: 1, batch: 1}}
+			steps := []helperTestStep{
+				{},
+				{getRestorePipeWriterArgExpect: "mock_1_1", getRestorePipeWriterResult: true, checkSkipFileArgTableOid: 1, checkSkipFileResult: false},
+			}
+
+			mockHelper := newHelperTest(oidBatch, steps)
+
+			// Prepare and write the toc file
+			testDir := "" //"/tmp/helper_test/20180101/20180101010101/"
+			*tocFile = fmt.Sprintf("%stest_toc.yaml", testDir)
+			writeTestTOC(*tocFile)
+			defer func() {
+				_ = os.Remove(*tocFile)
+			}()
+
+			*dataFile = "test_data.dat"
+			// Call the function under test
+			err := doRestoreAgentInternal(mockHelper, mockHelper)
+
+			Expect(err).ToNot(HaveOccurred())
+		})
+		It("successfully restores using multiple data files when inputs are valid and no errors occur", func() {
+			// Call the function under test
+			oidBatch := []oidWithBatch{{oid: 1, batch: 1}}
+			steps := []helperTestStep{
+				{},
+				{getRestorePipeWriterArgExpect: "mock_1_1", getRestorePipeWriterResult: true, checkSkipFileArgTableOid: 1, checkSkipFileResult: false},
+			}
+
+			mockHelper := newHelperTest(oidBatch, steps)
+
+			err := doRestoreAgentInternal(mockHelper, mockHelper)
+
+			Expect(err).ToNot(HaveOccurred())
+		})
+		It("skips batches with corresponding skip file in doRestoreAgent", func() {
+			// Test Scenario 1. Simulate 1 pass for the doRestoreAgent() function with the specified oids, batches and expected calls
+			oidBatch := []oidWithBatch{
+				{100, 0},
+				{200, 0},
+				{200, 1},
+				{200, 2},
+			}
+
+			expectedScenario := []helperTestStep{
+				{},                               // placeholder as steps start from 1
+				{"mock_100_0", true, -1, false},  // Can open pipe for table 100, check_skip_file shall not be called
+				{"mock_200_0", true, -1, false},  // Can open pipe for table 200, check_skip_file shall not be called
+				{"mock_200_1", false, 200, true}, // Can not open pipe for table 200, check_skip_file shall called, skip file exists
+				{"mock_200_2", true, -1, false},  // Went to the next batch, Can open pipe for table 200, check_skip_file shall not be called
```

Comment:
Maybe add these comments as an argument to the structure and then print them to the log at some level?

Reply:
I am trying to keep helperTestStep structure shorter and tests smaller.

Reply:
But this way it will be much more clear

Reply:
Can not agree with such categorical statement. I find comments are clearer and less messy.

Reply:
@whitehawk, what do you think about this?

Reply:
I would agree here with @RekGRpth, as printing some usefull and specific log will help to speed up the debug in case the test fails at some point.